core-v1/src/main/scala/engine/StandardEngine.scala

```diff
@@ -269,6 +270,32 @@ object StandardEngine {
       ExprResult.ValuesOfType(expr, finalState)
     }
 
+    override def visitWhen[I, B : ExtractValue.AsBoolean, O : OP](
+      expr: Expr.When[I, B, O, OP],
+    ): PO <:< I => ExprResult[PO, I, O, OP] = { implicit evPOisI =>
+      val initFalseResults: Seq[ExprResult[PO, I, B, OP]] =
+        new Array[ExprResult[PO, I, B, OP]](expr.conditionBranches.length).toSeq
```

Comment:
This isn't helpful, it creates an Array that must be copied on every write. If I'm gonna do this, I should use a `Vector`, so that the final product is a persistent immutable data structure that doesn't need to be copied again at the end.